flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -64,10 +64,15 @@
 
 	/**
 	 * The constant to use for the parallelism, if the system should use the number
-	 *  of currently available slots.
+	 * of currently available slots.
 	 */
 	public static final int PARALLELISM_AUTO_MAX = Integer.MAX_VALUE;
 
+	/**
+	 * The flag value indicating an unknown or unset parallelism.
+	 */
+	public static final int PARALLELISM_UNKNOWN = -1;
```

Comment:
The parallelism may be overridden elsewhere and we are not resetting it back to the default. This flag simply indicates a value for which the parallelism will not be overridden.